pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -874,27 +879,8 @@ func (m *kubeGenericRuntimeManager) computePodActions(ctx context.Context, pod *
 		}
 	}
 
-	// Check initialization progress.
-	initLastStatus, next, done := findNextInitContainerToRun(pod, podStatus)
-	if !done {
-		if next != nil {
-			initFailed := initLastStatus != nil && isInitContainerFailed(initLastStatus)
-			if initFailed && !shouldRestartOnFailure(pod) {
```

Comment:
comment to simplify the review:

this logic is moved to `computeInitContainerActions` so it will be all in one place